packages/teams-js/src/public/interfaces.ts

```diff
@@ -169,6 +169,71 @@ export enum FileOpenPreference {
   Web = 'web',
 }
 
+/**
+ * Action Types
+ */
+export enum ActionObjectType {
+  M365Content = 'm365content',
+}
+
+/**
+ * Data pertaining to object(s) the action is being performed on
+ */
+export interface BaseActionObject<T extends ActionObjectType> {
+  type: T;
```

Comment:
I believe this `type: T` property may also need its own doc comment, not 100% sure though.